pkg/types/relayer.go

```diff
@@ -78,33 +78,34 @@ type NodeStatus struct {
 	State   string
 }
 
-// ChainService is a sub-interface that encapsulates the explicit interactions with a chain, rather than through a provider.
-type ChainService interface {
-	Service
-
+type ChainReader interface {
+	// NewContractReader returns a new ContractReader.
+	// The format of contractReaderConfig depends on the implementation.
+	NewContractReader(ctx context.Context, contractReaderConfig []byte) (ContractReader, error)
 	// LatestHead returns the latest head for the underlying chain.
 	LatestHead(ctx context.Context) (Head, error)
 	// GetChainStatus returns the ChainStatus for this Relayer.
 	GetChainStatus(ctx context.Context) (ChainStatus, error)
 	// ListNodeStatuses returns the status of RPC nodes.
 	ListNodeStatuses(ctx context.Context, pageSize int32, pageToken string) (stats []NodeStatus, nextPageToken string, total int, err error)
+}
+
+// ChainService is a sub-interface that encapsulates the explicit interactions with a chain, rather than through a provider.
+type ChainService interface {
+	Service
```

Comment:
Why is it OK to change this interface? Seems like it should include `ChainReader`:
```suggestion
type ChainService interface {
	Service
	ChainReader
```

Reply:
Sry, should've marked this as draft, I haven't given it too much thought yet, but I did assume that we want the end state to be ChainWriter and ChainReader without ChainService.